forest/bonsai/simulate_gps_data.py

```diff
@@ -157,12 +158,511 @@ def bounding_box(center: Tuple[float, float], radius: int) -> Tuple:
+    to create an instance of a person"""
+    def __init__(self,
+                 vehicle: Vehicle,
+                 main_occupation: Occupation,
+                 active_status: int,
+                 travelling_status: int,
+                 preferred_places: list):
+        """This function sets the personality attributes
+
+        Args:
+            vehicle used for distances and time of flights
+            main_occupation used for routine action in weekdays
+            active_status = 0-10
+                used for probability in free time to take an action
+                or stay home
+            travelling status = 0-10
+                used to derive amount of distance travelled
+            preferred_places  = [x1, x2, x3]
+                used to sample action when free time
+                where x1-x3 are amenities (str)
+        """
+        self.vehicle = vehicle.value
+        self.main_occupation = main_occupation.value
+        self.active_status = active_status
+        self.travelling_status = travelling_status
+        self.preferred_places = preferred_places
+
+
+@dataclass
+class Action:
+    """Class containing potential actions for a Person.
+    
+    Args:
+        action: str, 'p', 'p_night' or 'fpf' indicating pause, pause
+             for the night or flight-pause-flight
+        destination_coordinates: tuple, destination's coordinates
+        duration: list, contains [minimum, maximum] duration of pause in seconds
+        preferred_exit: str, exit code
+    """
+    action: str
+    destination_coordinates: Tuple[float, float]
+    duration: list
+    preferred_exit: str
+
+
+class Person:
+    """This class represents a person
+    whose trajectories we want to simulate."""
+
+    def __init__(
+        self,
+        home_coordinates: Tuple[float, float],
+        attributes: Attributes,
+        local_places: Dict,
+    ):
+        """This function sets the basic attributes and information
+        to be used of the person.
+
+        Args:
+            home_coordinates: tuple, coordinates of primary home
+            attributes: Attributes class, consists of various information
+            local_places: dictionary, contains overpass nodes
+                of amenities near house
+
+        """
+
+        self.home_coordinates = home_coordinates
+        self.attributes = attributes
+        # used to update preferred exits in a day if already visited
+        self.preferred_places_today = self.attributes.preferred_places.copy()
+        self.office_today = False
+        # this will hold the coordinates of paths
+        # to each location visited
+        self.trips: Dict[list] = {}
+
+        # if employed/student find a place nearby to visit
+        # for work or studies
+        # also set which days within the week to visit it
+        # depending on active status
+        if self.attributes.main_occupation != "":
+
+            if len(local_places[self.attributes.main_occupation]) != 0:
```

Comment:
To help with readability we can do something like:
```
places_near_main_occupation = local_places[self.attributes.main_occupation.value]
```
Please feel free to suggest a shorter name for the new variable.

Reply:
I replaced that with `main_occupation_locations`, since technically it represents the possible locations for an office or university not the places around the main occupation.